pkg/nuclide-clang-rpc/lib/ClangFlagsManager.js

```diff
@@ -200,6 +211,33 @@ export default class ClangFlagsManager {
     return data;
   }
 
+  async _getModifiedFlags(src: string, originalFlags: Array<string>): Promise<Array<string>> {
+    // Look for the project-wide flags
+    const projectFlagsDir = await fsPromise.findNearestFile(
+      PROJECT_CLANG_FLAGS_FILE,
+      nuclideUri.dirname(src),
+    );
+    let projectFlags = null;
+    if (projectFlagsDir != null) {
+      const projectFlagsFile = nuclideUri.join(projectFlagsDir, PROJECT_CLANG_FLAGS_FILE);
+      projectFlags = await this._loadProjectCompilerFlags(projectFlagsFile);
+    }
+
+    if (projectFlags == null) {
+      return originalFlags;
+    }
+
+    const extraCompilerFlags = projectFlags.extraCompilerFlags.flags || [];
+    const ignoredCompilerFlags = projectFlags.ignoredCompilerFlags.flags || [];
+
+    let finalFlags = originalFlags.slice(0);
```

Comment:
Nitpick: the slice is unnecessary. we can write this block in the functional style:

```
return originalFlags
  .filter(flag => ignoredCompilerFlags.indexOf(flag) === -1)
  .concat(extraCompilerFlags);
```